editor/plugins/animation_player_editor_plugin.h

```diff
@@ -52,6 +52,7 @@ class AnimationPlayerEditor : public VBoxContainer {
 	AnimationPlayerEditorPlugin *plugin = nullptr;
 	AnimationMixer *original_node = nullptr; // For pinned mark in SceneTree.
 	AnimationPlayer *player = nullptr; // For AnimationPlayerEditor, could be dummy.
+	Node *cached_player_root_node = nullptr;
```

Comment:
I think it is generally OK, but I don't understand why the pointer is cached.

Reply:
Sorry for the late reply! 

It is unsafe. My thought process was: Since a broken NodePath to the player root would be impossible to resolve, keeping a direct raw pointer to the current player's root can always work as long as its validity is checked before accessing it (2nd commit in this PR does this validity check). 

This pointer's lifetime should be the same as the current AnimationPlayer, so I declared it right below `AnimationPlayer *player = nullptr;`. 

Maybe hiding the raw pointer in a nested struct could alleviate its unsafety. Something like:

```
...
AnimationPlayer *player = nullptr; // For AnimationPlayerEditor, could be dummy.
struct PlayerCache {
private:
    Node* root_node = nullptr;
public:
    bool check_valid(AnimationPlayer* player) {
        NodePath cached_root_path = player->get_path_to(root_node);
        return root_node == player->get_node_or_null(cached_root_path);
    }

    Node* get_root_node(AnimationPlayer* player) {
        if (check_valid(player)) {
            return root_node;
        }
        return nullptr;
    }

    Node* set_root_node(Node* node) {
        root_node = node;
    }
} player_cache;
...
```

Reply:
Can you make it to keep the InstanceID and be able to fetch node from it as needed as we do elsewhere? Like:

```cpp
ObjectID *cached_player_root_node;
```
```cpp
Node *AnimationPlayerEditor::get_root_node() {
    return Object::cast_to<Node>(ObjectDB::get_instance(cached_player_root_node));
}
```

Reply:
I changed the code to use the instance ID instead of a raw pointer. Everything still works as expected. Thanks for the advice! You can check if everything looks good in the new commit.